libkmod/libkmod-builtin.c

```diff
@@ -33,29 +33,27 @@ struct kmod_builtin_info {
 	char *buf;
 };
 
-static bool kmod_builtin_info_init(struct kmod_builtin_info *info, struct kmod_ctx *ctx)
+static int kmod_builtin_info_init(struct kmod_builtin_info *info, struct kmod_ctx *ctx)
 {
 	char path[PATH_MAX];
 	FILE *fp = NULL;
 	const char *dirname = kmod_get_dirname(ctx);
 	size_t len = strlen(dirname);
 
-	if ((len + 1 + strlen(MODULES_BUILTIN_MODINFO) + 1) >= sizeof(path)) {
-		errno = ENAMETOOLONG;
+	if ((len + 1 + strlen(MODULES_BUILTIN_MODINFO) + 1) >= sizeof(path))
 		return false;
```

Comment:
Since return value is changed from bool to int, this should be `return -ENAMETOOLONG;`